nop saved tensor hooks during tracing in dynamo + aot_autograd

bdhirsh opened the pull request:
Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113417
* #114646



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

bdhirsh committed:
nop saved tensor hooks during tracing in dynamo + aot_autograd

[ghstack-poisoned]

albanD reviewed (approved):
SGTM but dynamo CI needs to be appeased

bdhirsh committed:
Update on "nop saved tensor hooks during tracing in dynamo + aot_autograd"

Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

bdhirsh committed:
Update on "nop saved tensor hooks during tracing in dynamo + aot_autograd"

Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

bdhirsh committed:
Update on "nop saved tensor hooks during tracing in dynamo + aot_autograd"

Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

bdhirsh committed:
Update on "nop saved tensor hooks during tracing in dynamo + aot_autograd"

Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

bdhirsh committed:
Update on "nop saved tensor hooks during tracing in dynamo + aot_autograd"

Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

bdhirsh committed:
Update on "nop saved tensor hooks during tracing in dynamo + aot_autograd"

Fixes https://github.com/pytorch/pytorch/issues/113263

I thought it was a bit annoying that I had to no-op the hooks both during fake tensor tracing in dynamo, and again during tracing in AOTAutograd, but I guess this is unavoidable.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]
